cmd/snap-confine/snap-confine.c

```diff
@@ -116,6 +131,9 @@ int main(int argc, char **argv)
 			debug
 			    ("skipping sandbox setup, classic confinement in use");
 		} else {
+			if (snap_name == NULL) {
```

Comment:
Can you not check this here. We should have enough checks in main early on to ensure this never happens. If those are missing add them early there.